taichi/transforms/whole_kernel_cse.cpp

```diff
@@ -95,13 +112,16 @@ class WholeKernelCSE : public BasicStmtVisitor {
   void visit(Stmt *stmt) override {
     if (!stmt->common_statement_eliminable())
       return;
+    // container_statement is no need to be CSE
+    if (stmt->is_container_statement())
+      return;
     // Generic visitor for all CSE-able statements.
     if (is_done(stmt)) {
-      visible_stmts_.back()[std::type_index(typeid(*stmt))].insert(stmt);
+      visible_stmts_.back()[operand_type_hash(stmt)].insert(stmt);
       return;
     }
     for (auto &scope : visible_stmts_) {
-      for (auto &prev_stmt : scope[std::type_index(typeid(*stmt))]) {
+      for (auto &prev_stmt : scope[operand_type_hash(stmt)]) {
```

Comment:
I think we can precalculate the hash value before the line `for (auto &scope : visible_stmts_) {` and avoid calling the function in the for loop like this:
```suggestion
      for (auto &prev_stmt : scope[stmt_hash_value]) {
```

Reply:
Yes! I will do it!